storageClient/storageClient_test.go

```diff
@@ -555,3 +555,277 @@ func TestListGroupsWithClientForClientWithGroupsReturnsSuccess(t *testing.T) {
+			if group.GroupID == groupID {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Fatalf("Failed to list correct Group: Response( %+v) GroupID: %+v \n", responseList, groupID)
+		}
+	}
+}
+func TestListGroupsWithClientForClientWithGroupsFilteredByGroupNameReturnsOneGroupSuccess(t *testing.T) {
+	registrationClient := accountClient.New(e3dbClients.ClientConfig{Host: cyclopsServiceHost})
+	queenClientInfo, createAccountResponse, err := e3dbTest.MakeE3DBAccount(t, &registrationClient, uuid.New().String(), cyclopsServiceHost)
+	if err != nil {
+		t.Fatalf("Error %s making new account", err)
+	}
+	queenClientInfo.Host = cyclopsServiceHost
+	accountToken := createAccountResponse.AccountServiceToken
+	queenAccountClient := accountClient.New(queenClientInfo)
+	registrationToken, err := e3dbTest.CreateRegistrationToken(&queenAccountClient, accountToken)
+	if err != nil {
+		t.Fatalf("error %s creating account registration token using %+v %+v", err, queenAccountClient, accountToken)
+	}
+
+	reg, ClientConfig, err := e3dbTest.RegisterClient(testCtx, ClientServiceHost, registrationToken, "name")
+	if err != nil {
+		t.Fatalf("Error registering Client %+v %+v %+v ", reg, err, ClientConfig)
+	}
+	ClientConfig.Host = cyclopsServiceHost
+	regClient := New(ClientConfig)
+
+	//Create A group
+	encryptionKeyPair, err := e3dbClients.GenerateKeyPair()
+	group := CreateGroupRequest{
+		Name:              "TestGroup1" + uuid.New().String(),
+		PublicKey:         encryptionKeyPair.Public.Material,
+		EncryptedGroupKey: encryptionKeyPair.Private.Material,
+	}
+	group2 := CreateGroupRequest{
+		Name:              "TestGroup1" + uuid.New().String(),
+		PublicKey:         encryptionKeyPair.Public.Material,
+		EncryptedGroupKey: encryptionKeyPair.Private.Material,
+	}
+
+	response, err := regClient.CreateGroup(testCtx, group)
+	if err != nil {
+		t.Fatalf("Failed to create group \n Group( %+v) \n error %+v", group, err)
+	}
+	if response.Name != group.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", group.Name, response.Name, group)
+	}
+	groupCreatedID1 := response.GroupID
+	response, err = regClient.CreateGroup(testCtx, group2)
+	if err != nil {
+		t.Fatalf("Failed to create group \n Group( %+v) \n error %+v", group, err)
+	}
+	if response.Name != group2.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", group.Name, response.Name, group)
+	}
+	groupCreatedID2 := response.GroupID
+	listRequest := ListGroupsRequest{
+		GroupNames: []string{group.Name},
+	}
+	responseList, err := regClient.ListGroups(testCtx, listRequest)
+
+	if err != nil {
+		t.Fatalf("Group LIST: Group was Not Found: %+v ", responseList)
+	}
+	for _, groupID := range []uuid.UUID{groupCreatedID1} {
+		var found bool = false
+		for _, group := range responseList.Groups {
+			if group.GroupID == groupID {
+				found = true
+				break
```

Comment:
same note about not breaking early and consider removing the outer single iteration loop on 745